plugins/collections/get.js

```diff
@@ -6,6 +6,42 @@ const schema = require('screwdriver-data-schema');
 const getSchema = schema.models.collection.get;
 const idSchema = joi.reach(schema.models.collection.base, 'id');
 
+/**
+ * Helper function to get last builds of pipelines
+ *
+ * @param {Pipeline} pipelines     - pipeline models to get last builds for
+ * @param {Factory}  eventFactory - factory for getting last event model
+ * @returns {Array}
+ */
+function getPipelinesHealth(pipelines, eventFactory) {
+    return pipelines
+        .filter(pipeline => !!pipeline)
+        .map((pipeline) => {
+            const result = Object.assign({}, pipeline.toJson());
+
+            if (!pipeline.lastEventId) {
+                return result;
+            }
+
+            return eventFactory.get(pipeline.lastEventId)
+                .then((event) => {
+                    if (!event) {
+                        return result;
+                    }
+
+                    return event.getBuilds()
+                        .then((builds) => {
+                            if (builds.length) {
+                                result.lastBuilds = builds.map(b => b.toJson());
```

Comment:
If I remember correctly we are going to show the status of the last single build, right?
Then why not just have the logic here to return the health for that build instead of returning all the builds for that event. Are you going to handle that in UI?